caffe2/python/onnx/tests/c2_ref_test.py

```diff
@@ -220,32 +314,69 @@ def test_gemm_conversion(self):
             op_names.append(op.type)
         self.assertEqual(op_names, ['Scale', 'Scale', 'MatMul', 'Add'])
 
-        # with shape info (that indicates C is 1D), gemm will be
-        # converted to FC
-        _, op_strs = backend.convert_node(node_def.SerializeToString(
-        ), [make_tensor_value_info("C", onnx.TensorProto.FLOAT, (1,)).SerializeToString()])
+        # opset7
+        # with B's and C's shape info, gemm will be
+        # converted to FC/FCTransposed
+        _, op_strs = backend.convert_node(node_def_transpose_b.SerializeToString(
+        ), [make_tensor_value_info("B", onnx.TensorProto.FLOAT, (3,2)).SerializeToString(),
+            make_tensor_value_info("C", onnx.TensorProto.FLOAT, (3,)).SerializeToString()],
+        7)
         op_names = []
         for s in op_strs:
             op = caffe2_pb2.OperatorDef()
             op.ParseFromString(s)
             op_names.append(op.type)
         self.assertEqual(op_names, ['Scale', 'Scale', 'FC'])
 
-        # or with broadcast, gemm will be converted to fc
-        node_def = make_node(
-            'Gemm',
-            ['A', 'B', 'C'],
-            ["Y"],
-            transB=True,
-            broadcast=1)
+        _, op_strs = backend.convert_node(node_def.SerializeToString(
+        ), [make_tensor_value_info("B", onnx.TensorProto.FLOAT, (2,3)).SerializeToString(),
+            make_tensor_value_info("C", onnx.TensorProto.FLOAT, (3,)).SerializeToString()],
+        7)
+        op_names = []
+        for s in op_strs:
+            op = caffe2_pb2.OperatorDef()
+            op.ParseFromString(s)
+            op_names.append(op.type)
+        self.assertEqual(op_names, ['Scale', 'Scale', 'FCTransposed'])
 
-        _, op_strs = backend.convert_node(node_def.SerializeToString())
+        # opset6 without broadcast
+        # even though with B's and C's shape info, gemm will be
+        # will be converted to matmul + add
+        _, op_strs = backend.convert_node(node_def.SerializeToString(
+        ), [make_tensor_value_info("B", onnx.TensorProto.FLOAT, (3,2)).SerializeToString(),
+            make_tensor_value_info("C", onnx.TensorProto.FLOAT, (3,)).SerializeToString()],
```

Comment:
This case may be incorrect, because C has to match A*B's dimensions. Let's set C's shape as (3,2) too.

Reply:
checked.